### PR TITLE
Refactoring rules and mail formatting

### PR DESCRIPTION
**Pull request by maukoquiroga**

### CODE REVIEW

File: lib/betagouvbot/mail_action.rb [hunk 1]
```diff
@@ -9,11 +9,11 @@ def initialize(client, mail)
     end
 
     def subject
-      @mail[:personalizations][0][:subject]
+      @mail['personalizations'][0]['subject']
```
Comment (maukoquiroga) [inline lib/betagouvbot/mail_action.rb]: Smell...

File: lib/betagouvbot/mailer.rb [hunk 2]
```diff
@@ -13,6 +13,8 @@ def call(warnings, rules)
         end
       end
 
+      private
+
       def email(urgency, context, rules)
         mail = rules[urgency][:mail].format(context)
```
Comment (maukoquiroga) [inline lib/betagouvbot/mailer.rb]: This should fail, but it doesn't. cc @Morendil

**Comment by Morendil** [comment]
I've added more commits but I think https://github.com/sgmap/betaGouvBot/pull/80/commits/0cea82d27cf532ef0652815c3071644e63e2e1aa and https://github.com/sgmap/betaGouvBot/pull/80/commits/e654a5abf93db6b45b16697556c808cd76d16e78 don't necessarily belong in this PR. Possibly we should merge this as it is now, and push further refactoring to a new PR. WDYT?

**Comment by Morendil** [comment]
We're green, and we have some time until the next date when we're sending notifications (june 15th). Let's merge and forge on with refactoring.

I would also like to improve our confidence that refactorings do not cause regressions when talking with external APIs such as SendGrid, OVH or Github, perhaps by providing endpoints for manual testing specifically that for instance will send out a test email or manipulate a mailing list set aside for testing purposes, etc. WDYT?

**Comment by Morendil** [comment]
Hmm, this was not actually mergeable - Mailer is still expecting the old-style `rules`, but as `mailer_spec.rb` feeds it just that, we're still green even though (I strongly suspect) production is broken. #83 fixes this but it worries me that we can get in that situation.

**Comment by maukoquiroga** [comment]
@Morendil Yep, that's why I added a comment "this should fail, but it doesn't".

**Comment by Morendil** [comment]
Haha, missed it - or rather I saw it but underestimated its significance.

**Comment by maukoquiroga** [comment]
Yeah, it is more common in a dynamic language than in a compiled one. That's when code review's importance comes in 😄 .

**Pull request merged**